src/app/decorations/qgsdecorationtitledialog.cpp

```diff
@@ -0,0 +1,142 @@
+
+  // placement
+  cboPlacement->addItem( tr( "Top left" ), QgsDecorationItem::TopLeft );
+  cboPlacement->addItem( tr( "Top center" ), QgsDecorationItem::TopCenter );
+  cboPlacement->addItem( tr( "Top right" ), QgsDecorationItem::TopRight );
+  cboPlacement->addItem( tr( "Bottom left" ), QgsDecorationItem::BottomLeft );
+  cboPlacement->addItem( tr( "Bottom center" ), QgsDecorationItem::BottomCenter );
+  cboPlacement->addItem( tr( "Bottom right" ), QgsDecorationItem::BottomRight );
+  cboPlacement->setCurrentIndex( cboPlacement->findData( mDeco.placement() ) );
+  spnHorizontal->setValue( mDeco.mMarginHorizontal );
+  spnVertical->setValue( mDeco.mMarginVertical );
+  wgtUnitSelection->setUnits( QgsUnitTypes::RenderUnitList() << QgsUnitTypes::RenderMillimeters << QgsUnitTypes::RenderPercentage << QgsUnitTypes::RenderPixels );
+  wgtUnitSelection->setUnit( mDeco.mMarginUnit );
+
+  // font settings
+  mButtonFontStyle->setDialogTitle( tr( "Title Label Text Format" ) );
+  mButtonFontStyle->setMapCanvas( QgisApp::instance()->mapCanvas() );
+  mButtonFontStyle->setTextFormat( mDeco.textFormat() );
+}
+
+QgsDecorationTitleDialog::~QgsDecorationTitleDialog()
+{
+  QgsSettings settings;
+  settings.setValue( QStringLiteral( "Windows/DecorationTitle/geometry" ), saveGeometry() );
+}
+
+void QgsDecorationTitleDialog::buttonBox_accepted()
+{
+  apply();
+  accept();
+}
+
+void QgsDecorationTitleDialog::buttonBox_rejected()
+{
+  reject();
+}
+
+void QgsDecorationTitleDialog::mInsertExpressionButton_clicked()
+{
+  QString selText = txtTitleText->textCursor().selectedText();
+
+  // edit the selected expression if there's one
+  if ( selText.startsWith( QLatin1String( "[%" ) ) && selText.endsWith( QLatin1String( "%]" ) ) )
+    selText = selText.mid( 2, selText.size() - 4 );
+
+  QgsExpressionBuilderDialog exprDlg( nullptr, selText, this, QStringLiteral( "generic" ), QgisApp::instance()->mapCanvas()->mapSettings().expressionContext() );
+
+  exprDlg.setWindowTitle( QObject::tr( "Insert Expression" ) );
+  if ( exprDlg.exec() == QDialog::Accepted )
+  {
+    QString expression = exprDlg.expressionText();
+    if ( !expression.isEmpty() )
+    {
+      txtTitleText->insertPlainText( "[%" + expression + "%]" );
+    }
+  }
+}
+
+void QgsDecorationTitleDialog::apply()
+{
+  mDeco.setTextFormat( mButtonFontStyle->textFormat() );
+  mDeco.mLabelText = txtTitleText->toPlainText();
+  mDeco.mBackgroundColor = pbnBackgroundColor->color();
+  mDeco.setPlacement( static_cast< QgsDecorationItem::Placement>( cboPlacement->currentData().toInt() ) );
+  mDeco.mMarginUnit = wgtUnitSelection->unit();
+  mDeco.mMarginHorizontal = spnHorizontal->value();
+  mDeco.mMarginVertical = spnVertical->value();
+  mDeco.setEnabled( grpEnable->isChecked() );
+  mDeco.update();
+}
+
+void QgsDecorationTitleDialog::showHelp()
+{
+  //QgsHelp::openHelp( QStringLiteral( "introduction/general_tools.html#title-label" ) );
```

Comment:
@DelazJ , should I a/ remove the help button from this dialog, or b/ uncomment this line and assume that by the time 3.6 is released, the general_tools.html doc will have a #title-label section?

Reply:
@nirvn Thanks for having taken this into account. 
I'm not sure of b/; it means that 3.4 is released and this feature gets quickly documented (there's no priority in doc, so no guarantee). So i'd say hide the button in GUI and let's add a comment in the issue report to enable it when doc is ready.